src/Kestrel.Core/Internal/Http/IHttpParser.cs

```diff
@@ -11,5 +11,9 @@ public interface IHttpParser<TRequestHandler> where TRequestHandler : IHttpHeade
         bool ParseRequestLine(TRequestHandler handler, in ReadOnlySequence<byte> buffer, out SequencePosition consumed, out SequencePosition examined);
 
         bool ParseHeaders(TRequestHandler handler, in ReadOnlySequence<byte> buffer, out SequencePosition consumed, out SequencePosition examined, out int consumedBytes);
+
+        bool ParseRequestLine(TRequestHandler handler, ref BufferReader<byte> reader);
+
+        bool ParseHeaders(TRequestHandler handler, ref BufferReader<byte> reader);
```

Comment:
Should the interface have 2 `ParseHeaders` methods?

Reply:
In the process of condensing to 1.